### ISSUE TITLE
Wrong prefix match 

### ISSUE THREAD

**Issue by sualko** [signalapp/libsignal-protocol-javascript#53]
https://github.com/signalapp/libsignal-protocol-javascript/blob/f5a838f1ccc9bddb5e93b899a63de2dea9670e10/src/helpers.js#L35-L47

I think someone tried here to implement a prefix match, but it should not work, because it's the same as

```
return a.substring(0, a.length) == b.substring(0, b.length);
```

Am I right, or did I overlook something? Are you interested in a pr?

**Comment by elsehow** [comment]
I am definitely up for a PR! 

In the PR, be sure to make a test that:

- Fails as it's written currently
- Explains why the test should pass
- Passes with your propose implementation

Thank you for reviewing the code!

**Comment by sualko** [comment]
The problem is, that I'm not sure if the intended behavior is that this is a prefix match. It would be awesome if a developer could give a hint.

**Comment by elsehow** [comment]
@sualko Indeed. Unfortunately, this code all comes from the libsignal-javascript project.
